ui/ui_v3/scripts/select_table.js

```diff
@@ -0,0 +1,327 @@
+// Functions for table range selection, getting list of headers and filter values from sheet
+
+/**
+ * Detect the table's range based on user's selected cells
+ *
+ * @return {Object} dataRange - The detected table range's details
+ * @return {string} dateRange.inputSheet - The name of the sheet containing detected table
+ * @return {string} dateRange.rangeString - The A1Notation of the range detected
+ * @return {number} dateRange.headerRow - The header row number of the range detected
+ * @return {string} dateRange.headerRange - The header range of the range detected
+ */
+function autoDetectTable() {
+  var sheet = SpreadsheetApp.getActiveSheet();
+  var range = sheet.getDataRange();
```

Comment:
this is the entire range part of the sheet right, use a better name as this could be confusing.

Reply:
After few second thoughts , how is preComputeTableRange for the function name ?

Reply:
Also add a comment that is triggered on launch , and why we do it this way.